src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -39,3 +51,64 @@ def create_query_generator(plan: SnowflakePlan) -> QueryGenerator:
         )
 
     return QueryGenerator(plan.session, snowflake_create_table_plan_info)
+
+
+def replace_child(
+    parent: PlanTreeNode, old_node: PlanTreeNode, new_node: Selectable
+) -> None:
+    """
+    Helper function to replace the child node in the plan.
+    """
+    if not parent._is_valid_for_replacement:
+        raise ValueError(f"parent node {parent} is not valid for replacement.")
+
+    if old_node not in getattr(parent, "children_plan_nodes", parent.children):
+        raise ValueError(f"old_node {old_node} is not a child of parent {parent}.")
+
+    if isinstance(parent, SnowflakePlan):
+        assert parent.source_plan is not None
+        replace_child(parent.source_plan, old_node, new_node)
+        return
+
+    if isinstance(parent, SelectStatement):
+        parent.from_ = new_node
+        return
+
+    if isinstance(parent, SetStatement):
+        parent._nodes = [
```

Comment:
also if the new child is not selectable, we will have to create a selectable for it.

Reply:
To convert a generic `PlanTreeNode` into a selectable, we would also have to add analyzer as an argument which I am not against. But I'm wondering if we can make the requirement for `new_child` more strict to only be a `SelectableEntity` since both in repeated subquery elimination and large query breakdown, we only replace a subtree with a `SelectableEntity`?